### ISSUE TITLE
need a small delay multi-box session launches

### ISSUE THREAD

**Issue by mwrich4** [TheCheatsrichter/Gw2_Launchbuddy#76]
It may just be that with GW2 on an SSD and 32G RAM installed that I am an exception.

However, I consistently have what appears to be a contention issue between multiple sessions of GW2 being launched from buddy.  The exception is from the gw2 exe and says "You do not have sufficient privileges to patch the file at this location.  Please log in as an administrator and try again."  If I restart the failed instance, it starts properly.

~~More Info.  Upon thinking of this, I realized my first instance is started from a shortcut using these parameters "-shareArchive -nopatchui -email "******" -password "******".  Then when I feel a need, I start gw2buddy and launch more sessions.  The issue with gw2 only happens when I try to add 2 or more sessions using gw2buddy AFTER starting the initial session from the shortcut.  Perhaps this is the root-cause.~~  **PS** I was able to duplicate the issue starting all sessions from buddy.

I thought a brief delay after spawning multiple clients might relieve the issue.

**Comment by ShiroiKuma** [comment]
I’m having an issue where none of the instances after the first one even attempt to load up, not sure where to start then manually though.

Interesting theory on the performance idea, using 64GB Ram, 36 Cores and nVME drives here too, perhaps a delay or forced slow down might be good. I’ll see if I can force that myself and compile locally.

**Comment by mwrich4** [comment]
I was using ( and still am ) version 1.0.5 when I reported this.  I will try a new version and report back.

**Comment by mwrich4** [comment]
I tried 1.7.  I wasn't able to get even 2 screens to start at once.  I did keep getting messages about excessive fixups. I'd get the fixup message starting 1.7 or 1.0.5 after trying 1.7.  Plus, switching between 1.0.5 and 1.7,  my account list was lost each time.  

Interestingly, after 'trying 1.7' a few times, now that I've reverted to 1.0.5, it seems to launch 4 windows at once successfully every time so far.  I suspect perhaps you've stumbled into a .DLL hell.  ( I DO remember trying at least one version sometime between 1.0.5 and 1.7 a while ago.)

I am still using Windows 7 pro, 32GB, SSD or regular HD.

**Comment by mwrich4** [comment]
I guess it was wishful thinking.  I went back to 1.0.5 and initially the performance seems better (happens less often).  I still get an occasional failure with 1.0.5, so far it seems less frequent.  It didn't seem like the new version added any new files other than the .exe.  Perhaps a registry setting is a bit different now after exposure to 1.7.  
I  believe I had tried other versions of buddy along the way.  If you let me know what to target, I can export some of the registry keys if that helps.

**Comment by KairuByte** [comment]
The reg keys related to LB are pretty transient. All they do is keep track of known Client instances.

Any version of LB save files won’t really change much either. They really just store account info and a handful of other things. 

The exe is self contained, everything it needs gets packed inside it with a weaver. The exe also won’t show many signs of change, but the UI and feature set does change.

**Comment by ShiftLeftLogical** [comment]
@mwrich4 - I had the "same" issue and at some point (between moving back and forth between 1.6.2 and the previous version) the "same" issue went away.

The issue was (in my case but the results you are describing were the exact same as I had experienced, except version numbers of LB) that the process which LB appeared to correctly be grabbing for GW2 had a handle list of size 0 and in turn could never find the mutex to kill as the code will normally loop over the handles and kill the mutex but there was not a list to loop over. I checked that the code had the right process ID by printing a debug msg and checking in sysinternals and sure enough it did, yet LB's handle list for the process got back a size 0 list.

"Making the issue disappear, sadly":
At one point, I at least rebooted my machine but I believe also switched versions, (maybe 3rd time, to 1.6.2) and started LB back up per usual in visual studio and could never get the bug to reproduce ... the "stupid" obtaining of the process's handle list code worked as intended ;p , always ... I have started LB hundreds of times now and never had an issue except after the the initial install and first LB run manifesting itself after restarting LB for the second time (ever) through the multiple version changes and one PC reboot. I have been lazy and not wiped out all configuration files and tried from scratch yet ... which I really need to do if I want to solve it or get a better ... handle ... on the handles. (Yeah that last sentence deserves a slap but I had to do it.)
